git {ref,}log calls: pass --no-show-signature

goxberry opened the pull request:
If the `git` configuration setting `log.showSignature` is set to `true`, then some `git log` and `git reflog` invocations performed by `git-machete` return GPG signature output that causes `git-machete` to fail. The failures appear to occur because GPG signatures are assumed not to appear in the output of those commands.

This pull request resolves that failure mode by passing the `--no-show-signature` flag to all `git log` and `git reflog` queries. (Passing that flag to `git reflog` queries is necessary because `git reflog` is porcelain for `git log` plus some additional flags.)

Fixes #1286.

goxberry commented:
Amended commit so heading is ≤ 50 characters.

goxberry commented:
The `--no-show-signature` flag for `git log` was added in `git` 2.10.0, apparently[^1]. That release was also when the `log.showSignature` config option was added, so there's no need to pass that flag for `git` versions preceding 2.10.0. Possibly worth trying `git -c log.showSignature=false` (or replace `false` with `0` for brevity) instead.

[^1]: Per the [`git` 2.10.0 release notes](https://github.com/gitster/git/blob/557ae147e6cdc9db121269b058c757ac5092f9c9/Documentation/RelNotes/2.10.0.txt).

goxberry commented:
An example reproducing the error is provided in #1286.

PawelLipski commented:
Wow that's a non-trivial one 😯 thanks!

I now realized that IntelliJ (specifically, git4idea plugin) always adds this flag to `git` invocations:

```
16:00:53.878: [git-machete-intellij-plugin] git -c credential.helper= -c core.quotepath=false -c log.showSignature=false merge --ff-only refs/remotes/origin/develop
```

If you could introduce 3 extra changes:

1. maybe just pass `-c log.showSignature=false` to all git commands (so put it into `_popen_git` and `_run_git` functions)? I assume it doesn't harm, and can prevent unexpected errors in some other commands
2. add a test case in `test_git_operations.py` that would've captured the error (i.e. that fails without your changes to `git_operations.py`)
3. and most importantly, credit yourself in RELEASE_NOTES.md ‼️

PawelLipski commented:
Ouch, I've rebased onto fresh `develop` out of habit, please `git reset --keep @{upstream}` or force-push 😅

goxberry commented:
> Wow that's a non-trivial one 😯 thanks!

You're welcome!

> If you could introduce 3 extra changes:
>
> 1. Maybe just pass `-c log.showSignature=false` to all git commands (so put it into `_popen_git` and `_run_git` functions)? I assume it doesn't harm, and can prevent unexpected errors in some other commands. Also, pls then add a short comment about the purpose of this flag where it's used.

Added in [8b5f17d](https://github.com/VirtusLab/git-machete/pull/1285/commits/8b5f17df251b1f4b56713cbb70f7ea8b06f1556b).

> 2. Add a test case in `test_git_operations.py` that would've captured the error (i.e. that fails without your changes to `git_operations.py`).

Added in [4507d07](https://github.com/VirtusLab/git-machete/pull/1285/commits/4507d07ed3e35ebda4e60d442f3e4cf365052459).

> 3. And most importantly, credit yourself in `RELEASE_NOTES.md` :bangbang:

Added in [29773c6](https://github.com/VirtusLab/git-machete/pull/1285/commits/29773c65a0b718d3a6a739a6f08c751c92d85fa2).

Thank you for reviewing, and for maintaining `git machete`!

PawelLipski commented on git_machete/git_operations.py:
```diff
@@ -630,7 +640,12 @@ def get_counterpart_branches() -> Iterator[str]:
 
         # The trailing '--' is necessary to avoid ambiguity in case there is a file called just exactly like one of the branches.
-        entries = utils.get_non_empty_lines(self._popen_git("reflog", "show", "--format=%gD\t%H\t%gs", *(all_branches + ["--"])).stdout)
+        entries = utils.get_non_empty_lines(self._popen_git(
```
Actually, if you could revert these lines (here and below) &mdash; I'd say that the single-line format is easier to read.

PawelLipski reviewed (approved):
Thanks!

PawelLipski commented:
Feel free to merge once the nitpick is fixed
